
// [START declaration]
syntax = "proto3";
package message;

message Message {
    string msg = 1;
    bytes data = 2;
}

enum Role {
    PLAYER_A = 0;
    PLAYER_B = 1;
    OBSERVER = 2;
}

enum Command {
    START = 0; // 开始tick
    STOP = 1;  // 停止tick
    STEP = 2;  // 手动tick一下
    RESET = 3; // reset环境
}

message ObservationRequest {
    Command command = 1;
}

message PlayerRequest {
    Role role = 1;
    bytes data = 2;
}

service Rts {
    rpc ConnectObserver(stream ObservationRequest) returns (stream Message) {}
    rpc ConnectPlayer(stream PlayerRequest) returns (stream Message) {}
}

